lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3174,9 +3183,11 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
   AdjointValue prepareSeedAdjoint() {
     auto *ret = cast<ReturnInst>(getOriginal().findReturnBB()->getTerminator());
     auto origRetLoc = ret->getOperand().getLoc();
-    auto *rvi = builder.createRetainValue(origRetLoc, seed,
-                                          builder.getDefaultAtomicity());
-    assert(rvi->getParent() == getAdjoint().getEntryBlock());
+    if (seed->getType().isReferenceCounted(getModule())) {
+      auto *rvi = builder.createRetainValue(origRetLoc, seed,
+                                            builder.getDefaultAtomicity());
+      assert(rvi->getParent() == getAdjoint().getEntryBlock());
```

Comment:
K, i accidentally checked in this assertion. Please help remove this and the auto variable :) Thanks!

Reply:
Done in https://github.com/apple/swift/pull/22472/commits/4f369a758688637ee7d888acd64d14ecbfab96b6.